dot/network/stream_manager.go

```diff
@@ -0,0 +1,78 @@
+package network
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+var cleanupStreamInterval = time.Minute
+
+type streamData struct {
+	lastReceivedMessage time.Time
+	stream              network.Stream
+}
+
+// streamManager tracks inbound streams and runs a cleanup goroutine every `cleanupStreamInterval` to close streams that
+// we haven't received any data on for the last time period. this prevents keeping stale streams open and continuously trying to
+// read from it, which takes up lots of CPU over time.
+type streamManager struct {
+	ctx           context.Context
+	streamDataMap *sync.Map //map[string]*streamData
```

Comment:
never used a `sync.Map` before.  Are there multiple goroutines trying to read this map concurrently? Potentially you may get better performance from using a standard map and `sync.RWMutex`.

Reply:
yep there are multiple goroutines that'll access it, I think `sync.Map` is optimized for this us case so would be preferred over a standard map and a mutex

Reply:
https://golang.org/pkg/sync/#Map this case would be use case 2